components/Login/Login.js

```diff
@@ -1,69 +1,28 @@
-import { useSession, signIn, signOut } from "next-auth/react";
+import { useSession, signIn } from "next-auth/react";
 import LoginIcon from "@/public/icons/LoginIcon.svg";
 import LogoutIcon from "@/public/icons/LogoutIcon.svg";
 import styled from "styled-components";
-import { useEffect, useState } from "react";
-import RegularButton from "../Buttons/RegularButton";
-import ButtonWrapper from "../Buttons/ButtonWrapper";
 
 export default function Login({
   position,
   changeShowLogOutDialog,
-  showLogOutDialog,
+  changeSubmenuMode,
+  changeFlashcardSelection,
+  handleToggleMenu,
 }) {
   const { data: session } = useSession();
 
-  const [isSigning, setIsSigning] = useState("");
-
-  function handleLogout() {
-    signOut();
-    changeShowLogOutDialog(false);
-  }
-
-  useEffect(() => {
-    if (isSigning === "in") {
-      handleTimedOutSignIn();
-    }
-
-    function handleTimedOutSignIn() {
-      const signInTimeout = setTimeout(() => {
-        signIn();
-        setIsSigning("");
-      }, 300);
-
-      return () => {
-        clearTimeout(signInTimeout);
-      };
-    }
-  }, [isSigning]);
-
-  useEffect(() => {
-    if (isSigning === "out") {
-      handleTimedOutLogOutDialog();
-    }
-
-    function handleTimedOutLogOutDialog() {
-      const signOutTimeout = setTimeout(() => {
-        changeShowLogOutDialog(true);
-        setIsSigning("");
-      }, 300);
-
-      return () => {
-        clearTimeout(signOutTimeout);
-      };
-    }
-  }, [isSigning, changeShowLogOutDialog]);
-
   if (session) {
     return (
       <>
         <StyledButton
           $position={position}
           onClick={() => {
-            if (position === "header") {
-              changeShowLogOutDialog(true);
-            } else if (position === "menu") {
-              setIsSigning("out");
+            changeShowLogOutDialog(true);
+            if (position === "menu") {
+              changeSubmenuMode("default");
+              changeFlashcardSelection("all");
+              handleToggleMenu();
             }
```

Comment:
I would advice to pack all of this into a prop called "onClick" with a default value of an empty function:
```js
function Login({onClick = () => {}})
```
from the outside you can then pass this extra behavior if you need it, you don't need to check if you are in the menu or somewhere else, just pass an onclick function or don`t:
```jsx
<Login onClick={() => {
    changeSubmenuMode("default");
    changeFlashcardSelection("all");
    handleToggleMenu();
} /> // for menu

<Login /> // for header
```
Then you can call onClick here, if nothing was passed as a prop, it will be a function that does nothing:

```suggestion
onClick()
```

Reply:
Ah great, that makes it much easier. I was wondering yesterday if there was a simpler solution, but I couldn't think of one. Thanks. I just tried it and it works. 😁